src/components/date-picker/DatePicker.tsx

```diff
@@ -702,13 +705,20 @@ class DatePicker extends React.Component<DatePickerProps, DatePickerState> {
             onChangeAttr = { onChange: noop };
         }
 
-        // "name" prop is required for pattern validation to be surfaced on form submit. See components/form-elements/form/Form.js
-        // "title" prop is shown during constraint validation as a description of the pattern
-        // See https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern#usability
-        const additionalAttrs =
-            isAccessible && !this.canUseDateInputType
-                ? { name, pattern: ISO_DATE_FORMAT_PATTERN.source, title: 'YYYY-MM-DD' }
-                : {};
+        let additionalAttrs;
+        if (isAccessible && this.canUseDateInputType) {
+            additionalAttrs = {
+                max: this.formatDisplayDateType(maxDate) || '9999-12-31',
```

Comment:
Would it make sense to add any unit tests for testing the max & min is set properly?

Reply:
Yup, let me look into this!